sdk/rust/build.rs

```diff
@@ -8,16 +8,19 @@ fn main() {
         format!("-{pre}")
     };
 
-    let output = Command::new("git")
-        .arg("rev-parse")
-        .arg("HEAD")
-        .output()
-        .expect("failed to execute `git`");
-
-    let commit = if output.status.success() {
-        String::from_utf8(output.stdout).unwrap()
-    } else {
-        panic!("`git` failed: {}", String::from_utf8_lossy(&output.stderr));
+    let commit = match Command::new("git").arg("rev-parse").arg("HEAD").output() {
+        Ok(output) => {
+            if output.status.success() {
+                String::from_utf8(output.stdout).unwrap()
+            } else {
+                // Spin is not a part of a Git worktree
+                String::from("unknown")
+            }
+        }
+        Err(_e) => {
+            // Git is not installed
+            String::from("unknown")
+        }
```

Comment:
```suggestion
let commit = Command::new("git")
        .arg("rev-parse")
        .arg("HEAD")
        .output()
        .filter(|o| o.status.success())
        .map(|o| String::from_utf8_lossy(output.stdout))
        .unwrap_or("unknown");
   
```

Reply:
Thanks @rylev. This is much more concise! I slightly modified the mapping to first map `Result<Output, Error>` to an option.